beacon_node/network/src/sync/sampling.rs

```diff
@@ -626,3 +635,257 @@ mod request {
+            }
+        }
+
+        fn grab_beacon_processor_event(&mut self) {
+            if let Ok(BeaconWorkEvent {
+                drop_during_sync: _,
+                work: Work::RpcVerifyDataColumn(_),
+            }) = self.beacon_processor_rx.try_recv()
+            {
+            } else {
+                panic!("Should have sent a RpcVerifyDataColumn event");
+            }
+        }
+    }
+
+    fn active_sampling_request() -> (TestRig, ActiveSamplingRequest<TestBeaconChainType>) {
+        let log = test_logger();
+        let harness = BeaconChainHarness::<EphemeralHarnessType<E>>::builder(E)
+            .default_spec()
+            .logger(log.clone())
+            .deterministic_keypairs(1)
+            .fresh_ephemeral_store()
+            .build();
+        let chain = harness.chain;
+        let globals = Arc::new(NetworkGlobals::new_test_globals(
+            Vec::new(),
+            &log,
+            chain.spec.clone(),
+        ));
+        let (network_beacon_processor, beacon_processor_rx) =
+            NetworkBeaconProcessor::null_for_testing(
+                globals.clone(),
+                chain.clone(),
+                harness.runtime.task_executor.clone(),
+                log.clone(),
+            );
+        let (network_tx, network_rx) = mpsc::unbounded_channel();
+        let cx = SyncNetworkContext::new(
+            network_tx,
+            Arc::new(network_beacon_processor),
+            chain,
+            log.new(o!("component" => "network_context")),
+        );
+
+        let rig = TestRig {
+            cx,
+            beacon_processor_rx,
+            network_rx,
+            log,
+        };
+
+        let block_root = Hash256::random();
+        let id = SamplingRequester::ImportedBlock(block_root);
+        let sampling_config = SamplingConfig::Default;
+        let active_sampling_request = ActiveSamplingRequest::<TestBeaconChainType>::new(
+            block_root,
+            id,
+            &sampling_config,
+            rig.log.clone(),
+            &rig.cx.chain.spec,
+        );
+
+        (rig, active_sampling_request)
+    }
+
+    fn resp<'a, I: Iterator<Item = &'a ColumnIndex>>(
+        column_indexes: I,
+    ) -> (DataColumnSidecarList<E>, Duration) {
+        let mut list = DataColumnSidecarList::new();
+        for i in column_indexes {
+            list.push(Arc::new(DataColumnSidecar {
+                index: *i,
+                column: DataColumn::<E>::empty(),
+                kzg_commitments: KzgCommitments::<E>::empty(),
+                kzg_proofs: KzgProofs::<E>::empty(),
+                signed_block_header: SignedBeaconBlockHeader {
+                    message: BeaconBlockHeader::empty(),
+                    signature: Signature::empty(),
+                },
+                kzg_commitments_inclusion_proof: FixedVector::default(),
+            }));
+        }
+        (list, Duration::default())
+    }
+
+    #[test]
+    fn test_active_sampling_request() {
```

Comment:
Can you write tests in this format?  https://github.com/sigp/lighthouse/blob/5a966874da03cd987fef1629f732898a2859ce87/beacon_node/network/src/sync/block_lookups/tests.rs#L2011

We recently avoided adding unit tests too close to impl for event-based tests